src/components/BorrowerProfile/ShareButton.vue

```diff
@@ -39,12 +60,55 @@ export default {
 			required: true
 		}
 	},
+	created() {
+		// This query is part of the header query and should be in the cache.
+		// It's also not critical that this is 100% accurate since most
+		// borrowers will get prompted to share via the share query param
+		const myBorrowedLoanQuery = this.apollo.readQuery({
+			query: gql`
+				query mostRecentBorrowedLoan {
+					my {
+						mostRecentBorrowedLoan {
+							id
+						}
+					}
+				}
+			`,
+		});
+		this.borrowedLoanId = myBorrowedLoanQuery?.my?.mostRecentBorrowedLoan?.id ?? null;
+		this.modifiedShareMessage = this.shareMessage;
+	},
 	computed: {
-		linkedInTitle() {
-			return `A loan for ${this.loan.name}`;
+		progressPercent() {
+			// percent as a whole number 0-100
+			const percent = this.loan?.fundraisingPercent ?? 0;
+			return Math.round(percent * 100);
+		},
+		amountRemaining() {
+			return `$${this.loan?.loanAmount - this.loan?.loanFundraisingInfo?.fundedAmount}`;
+		},
+		isBorrower() {
+			// If the loan id's match or the share query param is present, the user is the borrower
+			return this.loan?.id === this.borrowedLoanId || this.$route.query.share === 'true';
 		},
 		modalTitle() {
-			return `Help ${this.name} spread the word.`;
+			if (this.isBorrower) {
+				switch (true) {
```

Comment:
I'm not sure what I think of the `switch (true)` syntax over if/else, but it's technically fine.